src/DiffEqOperators.jl

```diff
@@ -14,11 +14,13 @@ include("diffeqscalar.jl")
 include("array_operator.jl")
 
 ### Derivative Operators
-include("derivative_operators/derivative_operator.jl")
-include("derivative_operators/upwind_operator.jl")
 include("derivative_operators/fornberg.jl")
+include("derivative_operators/upwind_operator.jl")
+include("derivative_operators/derivative_irreg_operator.jl")
+include("derivative_operators/derivative_operator.jl")
+include("derivative_operators/abstract_operator_functions.jl")
 include("derivative_operators/boundary_operators.jl")
 
 export DiffEqScalar, DiffEqArrayOperator
-export AbstractDerivativeOperator, DerivativeOperator, UpwindOperator
+export AbstractDerivativeOperator, DerivativeOperator, UpwindOperator, FiniteDifference
```

Comment:
Not such a fan of the name FiniteDifference. We should bikeshed that a little bit.

Reply:
I am very excited about this functionality, and would be happy to help get some tests created.

 Since I see the current `DerivativeOperator` as a special case of this, consider whether the naming should be more like
- `DerivativeOperator` to `RegularDerivativeOperator`
- `FiniteDifference` to `IrregularDerivativeOperator`

The thing I don't understand about the interface is how much of the grid it is keeping around, at it looked to me like it only kept the `dx` rather than the whole `x` grid.  

For example, creating it with `FiniteDifference{Float64}(2,2,dx,length(x),:Dirichlet0,:Dirichlet0,BC=(0.0,0.0))` how do you know the exact boundaries it is valid on?  Those numbers might become very useful when we look at boundary value problems which create affine operators.

When operators need to be composed together, I think it will be very important to be able to do checks on the boundaries/etc?

Reply:
I would suggest to name use the names ``RegularOperator`` and ``IrregularOperator``, I find the ``Derivative`` redundant, as they're both subtypes of ``AbstractDerivativeOperator``. You also don't say ``UpwindDerivativeOperator`` (Although I guess you could do that as well...).

For the second part I'm not sure I understand the problem.

Reply:
Since there are other non-FDM operators, it might make sense to call it `AbstractFDMOperator` with `RegularFDMOperator` and `IrregularFDMOperator`.

I don't think the boundary values are required because they can be enclosed.